### PR TITLE
add SkipLastStreamTransformer and skipLast extension

### PR DESCRIPTION
**Pull request by HannibalKcc**
``` dart
Stream.fromIterable([1, 2, 3, 4, 5])
    .skipLast(3)
    .listen(print); // prints 1, 2
```

### CODE REVIEW

File: lib/src/transformers/skip_last.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    queue.add(data);
+  }
+
+  @override
+  void addError(EventSink<T> sink, Object e, [StackTrace? st]) =>
+      sink.addError(e, st);
+
+  @override
+  void close(EventSink<T> sink) {
+    queue.take(queue.length - count).forEach(sink.add);
```
Comment (hoc081098) [inline lib/src/transformers/skip_last.dart:23]: Missing check if `queue.length - count` is zero or negative?

**Review by hoc081098 — CHANGES_REQUESTED** [review]

File: lib/src/transformers/skip_last.dart (hunk 1 above)
Comment (HannibalKcc) [reply lib/src/transformers/skip_last.dart]: Thank you,i added missing parts already.Check it again pls.

File: lib/src/transformers/skip_last.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  final int count;
+  final List<T> queue = <T>[];
+
+  @override
+  void add(EventSink<T> sink, T data) {
+    queue.add(data);
+  }
+
+  @override
+  void addError(EventSink<T> sink, Object e, [StackTrace? st]) =>
```
Comment (hoc081098) [inline lib/src/transformers/skip_last.dart:18]: I have recently updated on master branch. Please update it

Comment (HannibalKcc) [reply lib/src/transformers/skip_last.dart:18]: done~

**Review by hoc081098 — APPROVED** [review]
LGTM

**Pull request merged**